compiler/rustc_mir/src/transform/large_enums.rs

```diff
@@ -0,0 +1,224 @@
+                                Mutability::Not,
+                            );
+                            let alloc = tcx.intern_const_alloc(alloc);
+                            alloc_cache.insert(ty, alloc);
+                            // FIXME(jknodt) use entry API
+                            alloc_cache.get(ty).unwrap()
+                        };
+
+                        let tmp_ty = tcx.mk_ty(ty::Array(
+                            tcx.types.usize,
+                            Const::from_usize(tcx, num_variants),
+                        ));
+
+                        let new_local = patch.new_temp(tmp_ty, span);
+                        let store_live =
+                            Statement { source_info, kind: StatementKind::StorageLive(new_local) };
+
+                        let place = Place { local: new_local, projection: List::empty() };
+                        let constant_vals = Constant {
+                            span,
+                            user_ty: None,
+                            literal: ConstantKind::Val(
+                                interpret::ConstValue::ByRef { alloc, offset: Size::ZERO },
+                                tmp_ty,
+                            ),
+                        };
+                        let rval = Rvalue::Use(Operand::Constant(box (constant_vals)));
+
+                        let const_assign = Statement {
+                            source_info,
+                            kind: StatementKind::Assign(box (place, rval)),
+                        };
+
+                        // FIXME(jknodt) do I need to add a storage live here for this place?
+                        let discr_place = Place {
+                            local: patch.new_temp(tcx.types.usize, span),
+                            projection: List::empty(),
+                        };
+
+                        let store_discr = Statement {
+                            source_info,
+                            kind: StatementKind::Assign(box (
+                                discr_place,
+                                Rvalue::Discriminant(*rhs),
+                            )),
+                        };
+
+                        // FIXME(jknodt) do I need to add a storage live here for this place?
+                        let size_place = Place {
+                            local: patch.new_temp(tcx.types.usize, span),
+                            projection: List::empty(),
+                        };
+
+                        let store_size = Statement {
+                            source_info,
+                            kind: StatementKind::Assign(box (
+                                size_place,
+                                Rvalue::Use(Operand::Copy(Place {
+                                    local: discr_place.local,
+                                    projection: tcx
+                                        .intern_place_elems(&[PlaceElem::Index(size_place.local)]),
```

Comment:
`discr_place`? The discriminant type is not always `usize`, so that will be another thing to take into account before using it here.